### PR TITLE
Add Multi Resource Alert changes for Metric Alert API

### PR DESCRIPTION
**Pull request by sanjaypavan**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]

File: specification/monitor/resource-manager/microsoft.insights/stable/2018-03-01/metricAlert_API.json [hunk 1]
```diff
@@ -617,7 +639,47 @@
         }
       },
       "description": "Specifies a metric dimension."
-    }
+    },
+    "MetricAlertMultipleResourceMultipleMetricCriteria":{
+      "x-ms-discriminator-value": "Microsoft.Azure.Monitor.MultipleResourceMultipleMetricCriteria",
```
[the remark this answers is not in the document]
Comment (sanjaypavan) [reply specification/monitor/resource-manager/microsoft.insights/stable/2018-03-01/metricAlert_API.json]: Hello @jhendrixMSFT We use similar name for **MetricAlertSingleResourceMultipleMetricCriteria**  We need to keep that consistent for the new changes as well. Any issues for not having a short name?

[the remark this answers is not in the document]
Comment (jhendrixMSFT) [reply specification/monitor/resource-manager/microsoft.insights/stable/2018-03-01/metricAlert_API.json]: I was referring to the discriminator value, it generates a really long type name e.g. `MicrosoftAzureMonitorMultipleResourceMultipleMetricCriteria`.  Removing the `Microsoft.Azure.Monitor.` prefix shortens it nicely and is easier to read.

[the remark this answers is not in the document]
Comment (sanjaypavan) [reply specification/monitor/resource-manager/microsoft.insights/stable/2018-03-01/metricAlert_API.json]: You are correct @jhendrixMSFT, but I need to follow the existing pattern which the customer already knew. We already have a discriminator value which is Microsoft.Azure.Monitor.SingleResourceMultipleMetricCriteria. Hence, I declared this one in the same way.

[the remark this answers is not in the document]
Comment (jhendrixMSFT) [reply specification/monitor/resource-manager/microsoft.insights/stable/2018-03-01/metricAlert_API.json]: So `Microsoft.Azure.Monitor.MultipleResourceMultipleMetricCriteria` is the value sent over the wire, including the `.` characters?

**Review by jhendrixMSFT — COMMENTED** [review]
Switching my review to comments.

**Review by jhendrixMSFT — APPROVED** [review]

**Comment by jhendrixMSFT** [comment]
@AutorestCI regenerate azure-sdk-for-go

**Comment by jhendrixMSFT** [comment]
@AutorestCI regenerate azure-sdk-for-go

**Comment by jhendrixMSFT** [comment]
@AutorestCI regenerate azure-sdk-for-go

**Pull request merged**